src/client/lod_mapblock.cpp

```diff
@@ -20,6 +20,55 @@
 static constexpr u16 quad_indices_02[] = {0, 1, 2, 2, 3, 0};
 static const auto &quad_indices = quad_indices_02;
 
+// a binary search implementation of std::countr_zero
+u8 countr_zeros(u64 n) {
```

Comment:
it's smarter to delegate this to `__builtin_ctz`, because the compiler is unlikely to be able to tell wtf your code is doing and will compile all these inefficient ifs/loops.
for a fallback a simple bitwise loop suffices. the compiler might even recognize it and replace it with its own efficient implementation.